sqlglot/tokens.py

```diff
@@ -1033,15 +1033,19 @@ def _scan_bits(self) -> None:
         self._advance()
         value = self._extract_value()
         try:
-            self._add(TokenType.BIT_STRING, f"{int(value, 2)}")
+            # If `value` can't be converted to a binary, fallback to tokenizing it as an identifier
+            int(value, 2)
+            self._add(TokenType.BIT_STRING, value[2:])  # Drop the 0b
         except ValueError:
             self._add(TokenType.IDENTIFIER)
 
     def _scan_hex(self) -> None:
         self._advance()
         value = self._extract_value()
         try:
-            self._add(TokenType.HEX_STRING, f"{int(value, 16)}")
+            # If `value` can't be converted to a hex, fallback to tokenizing it as an identifier
+            int(value, 16)
+            self._add(TokenType.HEX_STRING, value[2:])  # Drop the 0x
```

Comment:
Is removing the first two characters valid for all cases? Aren't we getting lucky that the two current formats start with `0x` and `X'`?

Reply:
Yep, discussed in https://github.com/tobymao/sqlglot/pull/1552#discussion_r1185453439

Reply:
Btw `X'` is handled elsewhere, see `_scan_formatted_string`.